File: 프로그래머스/0/181894. 2의 영역/2의 영역.cs
using System;

public class Solution {
    public int[] solution(int[] arr) {
        int start = -1;
        int end = arr.Length;
        for(int i = 0 ; i < arr.Length;i++)
        {
            if(arr[i] == 2 && start == -1)
            {
                start = i;
                end = i;
            }
            else if(arr[i] == 2)
                end = i;
        }
        int[] answer;
        if(start == -1)
        {
            answer = new int[1]{-1};
        }
        else
        {
            answer = new int[end - start + 1];
            Array.Copy(arr,start,answer,0,end - start + 1);
        }
            
        
        return answer;
    }
}

File: 프로그래머스/0/181894. 2의 영역/2의 영역.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    for(int i = 0; i < arr.size(); i++)
    {
        if(arr[i] == 2)
            answer.push_back(i);
    }
    if(answer.size() == 0)
    {
        answer.push_back(-1);
        return answer;
    }
    else if(answer.size() == 1)
    {
        answer.clear();
        answer.push_back(2);
        return answer;
    }
    else
    {
        int maxindex = *max_element(answer.begin(), answer.end());
        int minindex = *min_element(answer.begin(), answer.end());
        answer.clear();
        for(int i = minindex; i <= maxindex; i++)
        {
            answer.push_back(arr[i]);
        }
    }
    
    return answer;
}